internal/endorser/endorser.go

```diff
@@ -26,18 +26,19 @@ import (
 	"net/url"
 	"os"
 
+	"github.com/project-oak/transparent-release/internal/verifier"
 	"github.com/project-oak/transparent-release/pkg/amber"
 	"github.com/project-oak/transparent-release/pkg/intoto"
 	slsa "github.com/project-oak/transparent-release/pkg/intoto/slsa_provenance/v0.2"
 )
 
-// GenerateEndorsement generates an endorsement statement for the given binary SHA256 digest, for
+// GenerateEndorsement generates an endorsement statement for the given reference, for
```

Comment:
hmm... this does not sound right now. We don't generate an Endorsement for a reference. 

```suggestion
// GenerateEndorsement generates an endorsement statement for the binary SHA256 digest in the given reference, for
```